index.d.ts

```diff
@@ -0,0 +1,12 @@
+type Loggable = string | number | {};
+
+interface ILog {
+    info(message: Loggable): void;
+    debug(message: Loggable): void;
+    warn(message: Loggable): void;
+    error(message: Loggable): void;
+    trace(message: Loggable): void;
+}
+
+declare function logger(source: string): ILog;
+export = logger;
```

Comment:
Unfortunately this is the only way I can define that the export itself is a function.